integration-libs/s4-service/order/components/cancel-service-order-headline/cancel-service-order-headline.component.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { CartOutlets } from '@spartacus/cart/base/root';
+import { OrderDetailsService } from '@spartacus/order/components';
+import { Observable, map } from 'rxjs';
+
+@Component({
+  selector: 'cx-cancel-service-order-headline',
+  templateUrl: './cancel-service-order-headline.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CancelServiceOrderHeadlineComponent {
+  protected orderDetailsService = inject(OrderDetailsService);
+
+  order$: Observable<any> = this.orderDetailsService.getOrderDetails().pipe(
+    map((order) => ({
+      ...order,
+      entries: (order.entries || []).filter(
+        (entry) => entry.product && entry.product.productTypes === 'SERVICE'
+      ),
+    }))
+  );
```

Comment:
```suggestion
  order$ = this.orderDetailsService.getOrderDetails().pipe(
    map((order) => ({
      ...order,
      entries: (order.entries || []).filter(
        (entry) => entry.product && entry.product.productTypes === 'SERVICE'
      ),
    }))
  );
```
When `map` operator imported from the good module, you can remove type from `order$` as it will be properly infered.